File: backend/app/schemas/task.py
from pydantic import BaseModel, Field
from datetime import datetime


class TaskCreate(BaseModel):
    title: str = Field(..., max_length=200)
    description: str | None = None
    owner_name: str | None = None
    executor_name: str | None = None
    executor_type: str | None = None
    executor_note: str | None = None
    related_files: list[dict] = []
    evidence_source: str | None = None
    evidence_summary: str | None = None
    acceptance_criteria: str | None = None
    risk_level: str | None = None


class TaskUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    owner_name: str | None = None
    executor_name: str | None = None
    executor_type: str | None = None
    executor_note: str | None = None
    related_files: list[dict] | None = None
    evidence_source: str | None = None
    evidence_summary: str | None = None
    acceptance_criteria: str | None = None
    risk_level: str | None = None
    version: int


class TaskResponse(BaseModel):
    id: int
    issue_id: int
    title: str
    description: str | None
    status: str
    owner_name: str | None
    executor_name: str | None
    executor_type: str | None
    executor_note: str | None
    related_files: list[dict]
    evidence_source: str | None
    evidence_summary: str | None
    acceptance_criteria: str | None
    risk_level: str | None
    activity_log: list[dict]
    version: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class TaskStatusChangeRequest(BaseModel):
    status: str
    version: int
    details: str | None = None


class TaskDraftRequest(BaseModel):
    intent: str
    workspace_id: int
    issue_id: int
    context: dict | None = None


class TaskDraftResponse(BaseModel):
    title: str
    description: str | None = None
    executor_type: str | None = None
    executor_name: str | None = None
    owner_name: str | None = None
    evidence_summary: str | None = None
    acceptance_criteria: str | None = None
    risk_level: str | None = None
    tags: list[str] = []

File: frontend/src/types/task.ts
import type { ExecutorType, RelatedFile, ActivityLogEntry } from './issue'

export type TaskStatus = 'todo' | 'doing' | 'done'
export type RiskLevel = 'low' | 'medium' | 'high'

export interface Task {
  id: number
  issue_id: number
  title: string
  description: string | null
  status: TaskStatus
  owner_name: string | null
  executor_name: string | null
  executor_type: ExecutorType | null
  executor_note: string | null
  related_files: RelatedFile[]
  evidence_source: string | null
  evidence_summary: string | null
  acceptance_criteria: string | null
  risk_level: RiskLevel | null
  activity_log: ActivityLogEntry[]
  version: number
  created_at: string
  updated_at: string
}

export interface TaskCreate {
  title: string
  description?: string
  owner_name?: string
  executor_name?: string
  executor_type?: ExecutorType
  executor_note?: string
  related_files?: RelatedFile[]
  evidence_source?: string
  evidence_summary?: string
  acceptance_criteria?: string
  risk_level?: RiskLevel
  tags?: string[]
}

export interface TaskUpdate {
  title?: string
  description?: string
  owner_name?: string
  executor_name?: string
  executor_type?: ExecutorType
  executor_note?: string
  related_files?: RelatedFile[]
  evidence_source?: string
  evidence_summary?: string
  acceptance_criteria?: string
  risk_level?: RiskLevel
  version: number
}

export interface TaskStatusChangeRequest {
  status: TaskStatus
  version: number
  details?: string
}

// V1.1 AI 智能创建
export interface TaskDraftRequest {
  intent: string
  workspace_id: number
  issue_id: number
  context?: {
    recent_messages?: string[]
    related_files?: RelatedFile[]
    evidence?: string[]
  }
}

export interface TaskDraftResponse {
  title: string
  description: string
  executor_type: ExecutorType
  executor_name: string
  owner_name: string
  evidence_summary: string
  acceptance_criteria: string
  risk_level: RiskLevel
  tags: string[]
}

export interface ExecutorRecommend {
  name: string
  type: ExecutorType
  score: number
}
